src/stream/src/executor/dispatch.rs

```diff
@@ -66,40 +67,71 @@ impl DispatchExecutorInner {
+                    self.metrics
+                        .actor_output_buffer_blocking_duration_ns
+                        .with_label_values(&[
+                            &self.actor_id_str,
+                            &self.fragment_id_str,
+                            dispatcher.dispatcher_id_str(),
+                        ])
+                        .inc_by(start_time.elapsed().as_nanos() as u64);
                 }
             }
             Message::Chunk(chunk) => {
                 self.metrics
                     .actor_out_record_cnt
-                    .with_label_values(&[&self.actor_id_str])
+                    .with_label_values(&[&self.actor_id_str, &self.fragment_id_str])
                     .inc_by(chunk.cardinality() as _);
                 if self.dispatchers.len() == 1 {
                     // special clone optimization when there is only one downstream dispatcher
+                    let start_time = Instant::now();
                     self.single_inner_mut().dispatch_data(chunk).await?;
+                    self.metrics
+                        .actor_output_buffer_blocking_duration_ns
+                        .with_label_values(&[
+                            &self.actor_id_str,
+                            &self.fragment_id_str,
+                            self.dispatchers[0].dispatcher_id_str(),
+                        ])
+                        .inc_by(start_time.elapsed().as_nanos() as u64);
                 } else {
                     for dispatcher in &mut self.dispatchers {
+                        let start_time = Instant::now();
                         dispatcher.dispatch_data(chunk.clone()).await?;
+                        self.metrics
+                            .actor_output_buffer_blocking_duration_ns
+                            .with_label_values(&[
+                                &self.actor_id_str,
+                                &self.fragment_id_str,
+                                dispatcher.dispatcher_id_str(),
+                            ])
+                            .inc_by(start_time.elapsed().as_nanos() as u64);
                     }
                 }
             }
             Message::Barrier(barrier) => {
                 let mutation = barrier.mutation.clone();
                 self.pre_mutate_dispatchers(&mutation)?;
                 for dispatcher in &mut self.dispatchers {
+                    let start_time = Instant::now();
                     dispatcher.dispatch_barrier(barrier.clone()).await?;
+                    self.metrics
+                        .actor_output_buffer_blocking_duration_ns
+                        .with_label_values(&[
+                            &self.actor_id_str,
+                            &self.fragment_id_str,
+                            dispatcher.dispatcher_id_str(),
+                        ])
+                        .inc_by(start_time.elapsed().as_nanos() as u64);
```

Comment:
This part is verbose, but I failed to refactor it with a function/lambda due to the borrow checker 😇

Reply:
Then what about a macro? 🥵

Reply:
A little bit overkilling 😄